apps/api/src/api/application-api.ts

```diff
@@ -105,3 +106,62 @@ export const getApplicationById = async ({ applicationId }: { applicationId: num
 
 	return result;
 };
+
+export const approveApplication = async ({
+  applicationId,
+  approverId,
+}: ApproveApplication): Promise<{
+  success: boolean;
+  message?: string;
+  errors?: string | Error;
+  data?: any;
+}> => {
+  try {
+    // Fetch application
+	const database = getDbInstance();
+	const service: ApplicationService = applicationService(database);
+	const result = await service.getApplicationById({ id:applicationId });
+
+	if (!result.success) {
+		return {
+			success: false,
+			message: 'Application not found.',
+			errors: 'ApplicationNotFound',
+		  };
+	}
+
+	const { state } = result.data;
+	if(state ===  ApplicationStates.APPROVED){
+		return {
+			success: false,
+			message: 'application is already approved.',
+			errors: 'ApprovalConflict',
+		  };
+	}
+
+    // Check if we can transition to "APPROVED" state
+    if (!canTransitionTo(state as ApplicationState, ApplicationStates.APPROVED)) {
+		return {
+		  success: false,
+		  message: 'Invalid state transition.',
+		  errors: 'InvalidState',
+		};
+	  }
+    
+    result.data.state = ApplicationStates.APPROVED;
+
+	const update = { state: ApplicationStates.APPROVED, approved_at: new Date()};
```

Comment:
This indentation is another example that makes me ask about Prettier, each file in this PR would be worth formatting